lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -81,6 +81,13 @@ class AwsCompileFunctions {
       newFunction.Properties.Description = functionObject.description;
     }
 
+    if (functionObject.environment) {
```

Comment:
This entire `if` block can be represented as:

`newFunction.Properties.Environment.Variables = Object.assign({}, this.serverless.service.provider.environment, functionObject.environment)`

- assign can take multiple arguments
- assign will ignore undefined things
(also this would allow specifying env in the service that applies to all functions; as it stands this code would only assign service-level environment variables if the function also defines them)(